### ISSUE TITLE
Replace serially-numbered configuration properties with array-valued properties

### ISSUE THREAD

**Issue by mwoodiupui** [DSpace/DSpace#8658]
**Describe the bug**
Not really a bug, but an annoyance and an opportunity for improvement.  We have several groups of configuration properties such as `webui.browse.index.N` which were invented before we adopted Commons Configuration.  These are a pain to manage and a pain to consult.  Nowadays we could simply list multiple instances of the key `webui.browse.index` and get them all, in order, as an array from `ConfigurationService.getArrayProperty()` instead of trawling through the keyspace by iteratively composing numbered keys until we generate a miss.  In most cases the numbers are insignificant.

In the case of **parallel** property groups (of which I am sorry to say there are a nonzero number) we really should look at moving these array-of-structures configurations into Spring, which can handle them properly.  In their current form these cases are particularly fragile both to manage and to parse.  These get my vote for the second most heinous abuse of Properties after `java.util.logging`.

**Comment by tdonohue** [comment]
Agreed, this would be nice.  However, it'd definitely take some careful code refactoring that'd need to happen in a major release (8.0, 9.0, etc).  Will pull this over to "Help Wanted" board for now until we can find a volunteer.

**Comment by mwoodiupui** [comment]
As no one has objected, I've assigned myself and will work on this.

**Comment by mwoodiupui** [comment]
We seem to have less of this than I remember from DSpace 1.x days.  Noting here what I've found today:

These appear to be all related through a common serial number:

- `webui.browse.index.<n> = <index name> : metadata : \`
- `webui.browse.index.<n> = <index name> : item : <sort option name> : (asc | desc)`
- `webui.browse.index.tagcloud.<n> = true | false`
- `webui.browse.metadata.show-freq.<n> = true | false`

This seems related to the above by index name.  Only serial because it has multiple values?

- `webui.browse.link.<n> = <index name>:<display column metadata>`

I suspect that this is a separate list:

- `webui.itemlist.sort-option.<n> = <option name> : \`
